### ISSUE TITLE
Templates are persistent (living templates) modifications to them are automatically applied to all consumers

### ISSUE THREAD

**Issue by shiyaashrf** [netmod-wg/template-reqs#7]
### Requirements

If there are references to the template, then the modified template data should get expanded for all its existing references.

### Non-Requirements

None.

**Comment by kwatsen** [comment]
If I understand correctly, this is what I've heard called "Living Templates".  That is, the template-association is "live" such that any modification to the template immediately impacts all "consumers" of the template.  Yes?

**Comment by shiyaashrf** [comment]
I dont recall hearing the term "Living Templates". But from your explanation, I think I meant the same requirement.

**Comment by kwatsen** [comment]
That was the term we used at Juniper.  It's not one used (yet) in the IETF.  I guess it relates to the template association being long-lived (not just applied once).

**Comment by shiyaashrf** [comment]
Ok. I guess we are inline on this.

**Comment by jsterne** [comment]
Is another way to say this simply that the template is expanded every time a commit occurs?

**Comment by kwatsen** [comment]
Indeed, it is expanded (in <intended>) every time a commit occurs, because <intended> is subject to validation.

That said, in some implementations, <intended> may be ephemeral.  In such cases, the templates are also expanded every time a client gets <intended>.  

So there's two cases when templates may be expanded.
